Sources/WordPressUI/Extensions/Gravatar/Gravatar.swift

```diff
@@ -30,6 +68,43 @@ public struct Gravatar {
 
         return true
     }
+
+    /// Returns the Gravatar URL, for a given email, with the specified size + rating.
+    ///
+    /// - Parameters:
+    ///     - email: the user's email
+    ///     - size: required download size
+    ///     - rating: image rating filtering
+    ///
+    /// - Returns: Gravatar's URL
+    ///
+    public static func gravatarUrl(for email: String,
+                                   defaultImage: GravatarDefaultImage? = nil,
+                                   size: Int? = nil,
+                                   rating: GravatarRatings = .default) -> URL? {
+        let hash = gravatarHash(of: email)
+        let targetURL = String(format: "%@/%@?d=%@&s=%d&r=%@",
+                               Defaults.baseURL,
+                               hash,
+                               defaultImage?.rawValue ?? GravatarDefaultImage.fileNotFound.rawValue,
+                               size ?? Defaults.imageSize,
+                               rating.stringValue())
+        return URL(string: targetURL)
+    }
+
+    /// Returns the gravatar hash of an email
+    ///
+    /// - Parameter email: the email associated with the gravatar
+    /// - Returns: hashed email
+    ///
+    /// This really ought to be in a different place, like Gravatar.swift, but there's
+    /// lots of duplication around gravatars -nh
+    private static func gravatarHash(of email: String) -> String {
+        return email
+            .lowercased()
+            .trimmingCharacters(in: .whitespaces)
+            .md5Hash()
```

Comment:
I think Gravatar switched to sha256; we should also update.

Reply:
Great catch! Updated 😄